### ISSUE TITLE
InsecureRequestWarning

### ISSUE THREAD

**Issue by Rustymage** [jamesoff/simplemonitor#287]
Good Evening,

I've moved my install to a new pi and I can't seem to shake this issue - even after following the link.

Perhaps someone has come across this and has a solution?

I'm out of my depth now I've tried the link...

Thanks In advance.

`home/pi/.local/lib/python2.7/site-packages/urllib3/connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings`

**Comment by jamesoff** [comment]
Yeah, things like this can be a bit tricky.

What version of Python are you using and how did you install dependencies? I see 2.7 in the error message. I'm trying to reproduce it using a Rasbian VM, and it's working ok for me:

![Screenshot 2019-04-14 at 13 08 35](https://user-images.githubusercontent.com/5112771/56092631-6ef32c80-5eb6-11e9-9107-e1b281446895.png)

I just did `pip install --user -r requirements.txt` after cloning the repo.

Can you share your Python version and output of `pip list` (or whatever is suitable for how/where you installed deps)?

**Comment by Rustymage** [comment]
A little more information:

- Running on a RPi 3B+
- Previously running on RPi 2 with no issues but setup so long ago I forgot how
- Installed, got errors, followed the links, more errors, Google'd', tried other stuff, had broken links, fixed broken links...
- Came here to ask
 
> What version of Python are you using

```
$ python --version
Python 2.7.13
```

> Can you share your Python version and output of pip list (or whatever is suitable for how/where you installed deps)?

```
- asn1crypto (0.24.0)
- certifi (2019.3.9)
- cffi (1.12.2)
- chardet (3.0.4)
- cryptography (2.6.1)
- enum34 (1.1.6)
- idna (2.8)
- ipaddress (1.0.22)
- keyring (10.1)
- keyrings.alt (1.3)
- pip (9.0.1)
- pyasn1 (0.1.9)
- pycparser (2.19)
- pycrypto (2.6.1)
- pygobject (3.22.0)
- pyOpenSSL (19.0.0)
- pyxdg (0.25)
- requests (2.21.0)
- RPi.GPIO (0.6.5)
- SecretStorage (2.3.1)
- setuptools (33.1.1)
- six (1.12.0)
- urllib3 (1.24.1)
- wheel (0.29.0)
```

> I'm trying to reproduce it using a Rasbian VM, and it's working ok for me

```
pi@PhoenixPi:~/jamesoff-simplemonitor-abbfb03 $ python monitor.py -v --loops=1
SimpleMonitor v1.7
--> Loading main config from monitor.ini
--> Loading monitor config from monitors.ini
Adding host monitor RouterPi: checking host xxx.xxx.x.xx1 is pingable
Adding http monitor Pihole_page: Checking that accessing http://xxx.xxx.x.xx1/admin/index.php returns HTTP/200 OK
Adding http monitor WakeonLan_page: Checking that accessing https://domain1.co.uk returns HTTP/200 OK
Adding http monitor Ombi: Checking that accessing https://domain2.co.uk returns HTTP/200 OK
Adding host monitor ZeroPi: checking host xxx.xxx.x.xx2 is pingable
Adding host monitor VPNPi: checking host xxx.xxx.x.xx3 is pingable
Adding host monitor ABServer: checking host xxx.xxx.x.xx4 is pingable
Adding host monitor SabreServer: checking host xxx.xxx.x.xx5 is pingable
Adding host monitor PiPlex: checking host xxx.xxx.x.xx6 is pingable
Adding host monitor TorrentPi: checking host xxx.xxx.x.xx7 is pingable
--> Loaded 10 monitors.

Adding logfile logger logfile
Adding db logger db
Adding dbstatus logger dbstatus
Adding slack alerter slack
()
--> Starting... (loop runs every 1800s) Hit ^C to stop
Ran out of loop counter, will stop after this one
Fail: WakeonLan_page (Requests exception while opening URL: HTTPSConnectionPool(host='domain1.co.uk', port=443): Max retries exceeded with url: / (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'ssl3_read_bytes', 'sslv3 alert handshake failure')],)",),)))
Passed: Pihole_page
Passed: RouterPi
Passed: PiPlex
/home/pi/.local/lib/python2.7/site-packages/urllib3/connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
Passed: Ombi
Fail: ABServer (Command '['ping', '-c1', '-W5', 'xxx.xxx.x.xx4']' returned non-zero exit status 1)
Passed: ZeroPi
Passed: TorrentPi
Passed: SabreServer
Passed: VPNPi
```

**Comment by jamesoff** [comment]
Ok, looks like we have (almost) the same versions of libraries installed (very minor differences in only a few instances which I can't see would cause the problem).

I can reproduce it by adding `verify_hostname=0` to my `http` monitor. Would you happen to have that set too? (I guess if the host you're testing has a self-signed cert?)

Not sure why the warning is showing up now when it wasn't before. Did you perhaps have something like `PYTHONWARNINGS="ignore:Unverified HTTPS request"` in your environment on the old host? That does suppress the warning for me. (See [this StackOverflow entry](https://stackoverflow.com/questions/27981545/suppress-insecurerequestwarning-unverified-https-request-is-being-made-in-pytho).

If having verification disabled is what's actually desired there, I can have a go at making the code better at suppressing the warning without needing that environment variable set.

**Comment by Rustymage** [comment]
I'll answer this in a bit - SD card failure, unrelated I believe.

**Comment by Rustymage** [comment]
Fixed - I had `verify_hostname` set to `false` when I didn't have signed certs (it worked then). I removed that line and the errors are gone.

I did follow that guide you linked but it didn't work - not an issue as it appears sorted.

Once again, I appreciate the speedy help!

SD card replaced and log2ram installed to reduce wear on the card - let us hope that doesn't happen again.

:)


Case closed!
